feat(conversion-goals): add conversion goals team setting

jabahamondes opened the pull request:
## Problem

We have multiple "conversion goals" defined cross the app, for example the marketing one, the revenue events, the custom analytics events. This aim to centralize them as team conversion goals. 

The idea is that the user can define any event, action, data warehouse event as a conversion goal and then select for the different use case from a dropdown of conversion goal. (this logic for the dropdown will be implement when migrating marketing analytics into this new setting)

This would be transform:
<img width="623" height="554" alt="image" src="https://github.com/user-attachments/assets/d6e6544c-f49b-4613-a9fb-1d2105f3d77a" />
into  this ->:
<img width="614" height="530" alt="image" src="https://github.com/user-attachments/assets/def8a370-bcca-484b-8414-38364227e5f5" />



## Changes

- Introduced ConversionGoal and ConversionGoalCategory types in schema.
- Added conversion_goals property to the Team model for managing conversion goals.
- Created ConversionGoalsSettings component for UI management of conversion goals.
- Implemented logic for adding, updating, and removing conversion goals.
- Updated settings map to include conversion goals section.
- Added tests for conversion goals validation.
- Putting this behind a FF

This feature allows users to define and manage conversion goals across various analytics.

## How did you test this code?
Manually:
<img width="850" height="449" alt="Screenshot 2025-12-18 at 6 23 41 PM" src="https://github.com/user-attachments/assets/0f47c9fd-8312-434f-a9fe-6449f8c817dd" />
<img width="687" height="430" alt="Screenshot 2025-12-18 at 6 23 37 PM" src="https://github.com/user-attachments/assets/ba491468-9879-4db8-8fb5-0e15b06c7f2e" />
Not showing without FF
<img width="289" height="273" alt="Screenshot 2025-12-18 at 5 51 15 PM" src="https://github.com/user-attachments/assets/67458986-cb1d-4324-b5dc-f27ce9d94b3c" />

jabahamondes committed:
feat(conversion-goals): add conversion goals feature with settings and logic

- Introduced ConversionGoal and ConversionGoalCategory types in schema.
- Added conversion_goals property to the Team model for managing conversion goals.
- Created ConversionGoalsSettings component for UI management of conversion goals.
- Implemented logic for adding, updating, and removing conversion goals.
- Updated settings map to include conversion goals section.
- Added tests for conversion goals validation.

This feature allows users to define and manage conversion goals across various analytics.

jabahamondes committed:
Greptile comments

github-actions[bot] committed:
test(storybook): update UI snapshots

rafaeelaudibert reviewed (commented):
This is cool! I wonder if this is the best place for it, though. What if it was inside data management instead?  
  
Will let @arthurdedeus review this as well, so not approving  
  
![image.png](https://app.graphite.com/user-attachments/assets/21202c9e-46d8-47ed-9108-5ea3422f5076.png)

arthurdedeus reviewed (commented):
I'm probably lacking context here, but one thing that I couldn't understand is how are we measuring conversion if the conversion goal object has a single event filter.

I feel like there should be 2 steps in order to get a "conversion" from one to another.

For acquisition, e.g.: how may users viewed your signup page (event 1) and completed signup (event 2). Otherwise it would only be a count of the isolated signups.

Am I missing something here?

jabahamondes commented on frontend/src/queries/schema/schema-general.ts:
```diff
@@ -4532,6 +4527,32 @@ export enum MarketingAnalyticsHelperForColumnNames {
+    /** Optional description */
+    description?: string
+    /** Category (acquisition, activation, retention, referral, revenue) */
+    category?: ConversionGoalCategory
```
Yeah, maybe not. I'm not 100% sure if these options cover all the categories though, I could put like `other` category.

jabahamondes committed:
Merge remote-tracking branch 'origin/master' into feat/creating-a-team-conversion-goal-entity

github-actions[bot] committed:
test(storybook): update UI snapshots

jabahamondes committed:
refactor: Replace Conversion Goals with Core Events

- Updated FEATURE_FLAGS to reflect the new core events feature.
- Removed ConversionGoal and ConversionGoalCategory from schema.json and schema-general.ts.
- Introduced CoreEvent and CoreEventCategory in schema.json and schema-general.ts.
- Added CoreEventsSettings component for managing core events.
- Updated DataManagementScene to include core events tab.
- Created coreEventsLogic for handling core events state.
- Added TeamCoreEventsConfig model and serializer for team-level core events configuration.
- Updated tests to validate core events functionality.

jabahamondes committed:
Merge branch 'feat/creating-a-team-conversion-goal-entity' of github.com:PostHog/posthog into feat/creating-a-team-conversion-goal-entity

github-actions[bot] committed:
test(storybook): update UI snapshots

jabahamondes committed:
feat: Integrate core events configuration into team settings

- Added core_events_config to TeamType and initialized it in MOCK_DEFAULT_TEAM.
- Updated CoreEventsSettings to handle core events with new DataWarehouseFilter type.
- Introduced coreEventsConfigurationDescriber for detailed change descriptions in team activity.
- Refactored extra_settings in TeamType to use a more generic Record type.

github-actions[bot] committed:
test(backend): update query snapshots

jabahamondes committed:
update snapshots

jabahamondes committed:
Merge remote-tracking branch 'origin/master' into feat/creating-a-team-conversion-goal-entity

jabahamondes committed:
Fixing mypybaseline

jabahamondes committed:
Merge branch 'feat/creating-a-team-conversion-goal-entity' of github.com:PostHog/posthog into feat/creating-a-team-conversion-goal-entity

jabahamondes committed:
feat: Add Core Events to product structure

- Introduced Core Events in products.json and products.tsx with relevant metadata.
- Updated routes in scenes.ts to include Core Events.
- Created initial manifest for Core Events in a new manifest.tsx file.
- Added an empty __init__.py file for core_events module.

jabahamondes commented:
@rafaeelaudibert @arthurdedeus , at the end, I created a new model as we dicussed it. I also moved the definition inside the schema (like actions). The idea is that I will then create the marketing usage of this. I will make the components to pick between the existing core events and all of that experience/logic.

This is all behind a ff btw.

<img width="758" height="594" alt="image" src="https://github.com/user-attachments/assets/92d61194-76e2-4f86-b18b-6c8b0206ff74" />

arthurdedeus commented on frontend/src/scenes/settings/environment/coreEventsLogic.ts:
```diff
@@ -0,0 +1,57 @@
+                core_events_config: {
+                    core_events: newEvents,
+                },
+            })
```
All that said, I think this is fine while we're still building the thing. Just something to keep in mind as it can be a source for bugs when we release it.

arthurdedeus reviewed (approved):
Left a comment regarding the API/CRUD. No need to fix rn, but good to keep in mind!
